client/client.go

```diff
@@ -261,53 +259,64 @@ type httpClusterClient struct {
 	selectionMode EndpointSelectionMode
 }
 
-func (c *httpClusterClient) getLeaderEndpoint() (string, error) {
-	mAPI := NewMembersAPI(c)
-	leader, err := mAPI.Leader(context.Background())
+func (c *httpClusterClient) getLeaderEndpoint(ctx context.Context, eps []url.URL) (string, error) {
+	// To perform a lookup on the new endpoint list without using the current
+	// client, we'll copy it
+	clientCopy := &httpClusterClient{
+		clientFactory: c.clientFactory,
+		credentials:   c.credentials,
+		rand:          c.rand,
+
+		pinned:    0,
+		endpoints: eps,
```

Comment:
Technically this slice of eps (`neps` from the caller) is used both for the MembersAPI client and for this client in the near future, so it's owned by two different clients.
I get that their scopes don't overlap because MembersAPI isn't used for very long, but still seems a little scary.. especially if MembersAPI decides to make any changes to its client endpoints.